### ISSUE TITLE
Wherigo: cgeo closes if after show locations on map the first button touched is Wherigo on the navigation menu

### ISSUE THREAD

**Issue by rouroni1** [cgeo/cgeo#16327]
### Describe your problem!

After select to map the locations of a cartridge, if the first button I touch is the Wherigo Icon in the navigation menu, cgeo closes.
If I press home first or any other icon, if I press the Wherigo icon, it goes to the wherigo player like it should.

https://github.com/user-attachments/assets/1e6a3f78-ae85-4e38-8ca8-c94238a8d37b

https://github.com/user-attachments/assets/3bb50813-e3f8-4f3c-93a5-9ec1f580b404

### How to reproduce?

Tested with various cartridges and got the same result.

1. Open a cartridge
2. Click on the map icon to show in the map the locations
3. Click the Wherigo icon on the navigation menu at the bottom right
4. cgeo closes, like if I press the Android home button

But if I press any other bottom icon, and after the wherigo icon, then it goes to the wherigo player as it should.

### Actual result after these steps?

cgeo closes, but after opening the wherigo is still running. It's like I've pressed Android home button, but I didn't.

### Expected result after these steps?

If the wherigo button is pressed, it should go to the wherigo player menu.

### Reproducible

Yes

### c:geo Version

2024.11.05-NB-daa75a4

**Comment by eddiemuc** [comment]
COnfirmed and reproducable. Logcat shows no stack trace but this ANR message:

```
2024-11-09 10:19:57.725   742-3406  Acti...ger system_server    E  ANR in cgeo.geocaching.developer (cgeo.geocaching.developer/cgeo.geocaching.wherigo.WherigoActivity)
                                                                   PID: 3064
                                                                   Reason: Input dispatching timed out (8f72da5 cgeo.geocaching.developer/cgeo.geocaching.wherigo.WherigoActivity (server) is not responding. Waited 5000ms for FocusEvent(hasFocus=false))
                                                                   Parent: cgeo.geocaching.developer/cgeo.geocaching.wherigo.WherigoActivity
                                                                   ErrorId: 4eb2b902-2adb-4718-9789-918f802aaf4b
                                                                   Frozen: false
                                                                   Load: 8.13 / 6.89 / 4.77
                                                                   ----- Output from /proc/pressure/memory -----
                                                                   some avg10=13.99 avg60=18.36 avg300=13.38 total=286590458
                                                                   full avg10=7.90 avg60=8.64 avg300=5.49 total=130172079
                                                                   ----- End output from /proc/pressure/memory -----

```

**Comment by eddiemuc** [comment]
Fixed with PR #16330 by changing the launchMode to singleTop.

It might be noted that the problem only did appears with map activity, and it does not appear if the map is given some time to do all the stuff it seems to do on its launch (which seems to be quite a lot). So root cause here might be that the map is doing too much stuff on main thread on startup. This might be something to look into as part of unified map activities.
